File: programmers/string/babblings/babblings.py
def is_valid(word):
    valid_strings = ["aya", "ye", "woo", "ma"]
    for valid in valid_strings:
        word = word.replace(valid, ' ')
    return word.strip() == ''

def solution(babbling):
    valid_words = [word for word in babbling if is_valid(word)]

    return len(valid_words)

File: programmers/string/babblings/babblings.ts
const isValidWord = (word: string): boolean => {
  const validSubstrings = ["aya", "ye", "woo", "ma"];

  let tempWord = word;
  for (let validSubstring of validSubstrings) {
    tempWord = tempWord.split(validSubstring).join("");
  }
  return tempWord === "";
};

function solution(words: string[]): number {
  return words.filter((word) => isValidWord(word)).length;
}
